### DISCUSSION TITLE
One-to-One relationships raise exceptions when `configure_mappers()` is called

### DISCUSSION THREAD

**Discussion by jgadling** [sqlalchemy/sqlalchemy#10943] [Usage Questions] [answered]
### Describe the bug

Per the documentation SQLAlchemy supports one-to-one relationships, and it will allow you to define them, however the mapper raises an exception when trying to define two many-to-one relationships (which is what the docs imply this configuration is doing behind the scenes)

### Optional link from https://docs.sqlalchemy.org which documents the behavior that is expected

https://docs.sqlalchemy.org/en/20/orm/basic_relationships.html#one-to-one

### SQLAlchemy Version in Use

SQLAlchemy==2.0.25

### DBAPI (i.e. the database driver)

n/a

### Database Vendor and Major Version

n/a

### Python Version

3.11.4

### Operating system

Linux, OSX

### To Reproduce

```python
from sqlalchemy import ForeignKey, MetaData
from sqlalchemy.orm import (DeclarativeBase, Mapped, configure_mappers,
                            mapped_column, relationship)

meta = MetaData()


class Base(DeclarativeBase):
    metadata = meta


class Parent(Base):
    __tablename__ = "parent_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    child: Mapped["Child"] = relationship(back_populates="parent")


class Child(Base):
    __tablename__ = "child_table"

    id: Mapped[int] = mapped_column(primary_key=True)
    parent_id: Mapped[int] = mapped_column(ForeignKey("parent_table.id"))
    parent: Mapped["Parent"] = relationship(back_populates="child")


configure_mappers()
```

### Error

```
Traceback (most recent call last):
  File "/Users/jgadling/code/test.py", line 81, in <module>
    configure_mappers()
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/mapper.py", line 4168, in configure_mappers
    _configure_registries(_all_registries(), cascade=True)
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/mapper.py", line 4199, in _configure_registries
    _do_configure_registries(registries, cascade)
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/mapper.py", line 4240, in _do_configure_registries
    mapper._post_configure_properties()
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/mapper.py", line 2403, in _post_configure_properties
    prop.init()
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/interfaces.py", line 579, in init
    self.do_init()
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/relationships.py", line 1643, in do_init
    self._generate_backref()
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/relationships.py", line 2129, in _generate_backref
    self._add_reverse_property(self.back_populates)
  File "/opt/homebrew/lib/python3.11/site-packages/sqlalchemy/orm/relationships.py", line 1610, in _add_reverse_property
    raise sa_exc.ArgumentError(
sqlalchemy.exc.ArgumentError: ConsensusGenome.metrics and back-reference MetricConsensusGenome.consensus_genome are both of the same direction <RelationshipDirection.MANYTOONE: 2>.  Did you mean to set remote_side on the many-to-one side ?
``` 

### Additional context

I'm not sure if this functionality is specifically not supported with more interesting configurations that require the use of an explicit `configure_mappers()` call, but I wanted to see if this is the expected behavior, or if there's a way to get the mapper to play nice.

**Comment by jgadling** [comment]
Sorry - I mixed up my test scripts - I'll post an updated version shortly.

**Comment by zzzeek** [comment, marked as answer]
hi -

OK, though I can likely already point you to guidance for this errorL

     sqlalchemy.exc.ArgumentError: ConsensusGenome.metrics and back-reference MetricConsensusGenome.consensus_genome are both of the same direction <RelationshipDirection.MANYTOONE: 2>.  Did you mean to set remote_side on the many-to-one side ?

Based on that error message, we would assume you have a class ConsensusGenome and I would gather a subclass of it called MetricConsensusGenome, and then you have a relationship or two between them, which would be self-referential, that is, the adjacency list pattern.  to set many-to-one with adjacency list requires the use of the remote_side argument, a tutorial for this is at https://docs.sqlalchemy.org/en/20/orm/self_referential.html#self-referential

**Reply by jgadling** [reply]
Thanks for the feedback, and for being a rubber duck. The actual problem in our case was that we actually have two separate tables in a parent-child relationship (not set up to be self-referencing) but (because lack of coffee) were defining a `related_table_id` field in each table to refer to the other, when we actually only needed to set up the relationship in one direction.
